presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java

```diff
@@ -143,7 +138,7 @@ public HiveMetadata get()
     {
         SemiTransactionalHiveMetastore metastore = new SemiTransactionalHiveMetastore(
                 hdfsEnvironment,
-                CachingHiveMetastore.memoizeMetastore(this.metastore, perTransactionCacheMaximumSize), // per-transaction cache
```

Comment:
Why do we need to remove this per-transaction cache ?

Reply:
@Praveen2112 Because this.metastore used to be evaluated as GlueHiveMetastore and everytime a new cache map was being created because of memoizeMetastore so it never read from cache, it always used to boil down to: io.prestosql.plugin.hive.metastore.glue.GlueHiveMetastore#getTable

Reply:
@gs-asrivastav But after we bind `CachingHiveMetastore` to `GlueHiveMetastore` it should point to `CachingHiveMetastore`. And the `CachingHiveMetastore` which we get from  `memoizeMetastore` will now acts as second layer of cache for our existing `Metastore`.  IIRC this second layer of cache is used to ensure consistent state for tables involved in that transaction are maintained. (In case of a non caching metastore which happens if we didn't set this config `hive.metastore-cache-ttl` or set it to 0ms)

Reply:
@Praveen2112 is correct. This is used for both consistency and performance for multiple calls for the same data within the same transaction (or query). The engine may ask for the same data multiple times (perhaps across different metadata calls), and we don't want to go to the metastore for each one (especially because the results could change).

Reply:
let me verify this with the binding changes made in ’GlueMetastoreModule‘, I’ll revert this change if gluestore is not called everytime for fetching table metadata.

Reply:
@Praveen2112 @electrum reverted this change.